src/cli/server/server_config.rs

```diff
@@ -1,30 +1,37 @@
 use std::net::{IpAddr, Ipv4Addr, SocketAddr};
+use std::ops::Deref;
 use std::sync::Arc;
 
-use crate::blueprint::{Blueprint, Http};
-use crate::cli::runtime::init;
-use crate::http::AppContext;
+use crate::blueprint::Http;
+use crate::builder::TailcallExecutor;
 
 pub struct ServerConfig {
-    pub blueprint: Blueprint,
-    pub app_ctx: Arc<AppContext>,
+    pub tailcall_executor: TailcallExecutor,
 }
 
 impl ServerConfig {
-    pub fn new(blueprint: Blueprint) -> Self {
-        let server_context = Arc::new(AppContext::new(
-            blueprint.clone(),
-            init(&blueprint.upstream, blueprint.server.script.clone()),
-        ));
-        Self { app_ctx: server_context, blueprint }
+    pub fn new(mut tailcall_executor: TailcallExecutor) -> Self {
+        let blueprint = tailcall_executor.app_ctx.blueprint.clone();
+        let rt = crate::cli::runtime::init(
+            &blueprint.upstream,
+            tailcall_executor.app_ctx.blueprint.server.script.clone(),
+        );
+        let mut app_ctx = (*tailcall_executor.app_ctx.deref()).clone();
+        app_ctx.runtime = rt;
```

Comment:
should not set the runtime within the executor. This will have side-effects somewhere else.